internal/controller/frrconfiguration_controller.go

```diff
@@ -89,9 +111,66 @@ func (r *FRRConfigurationReconciler) applyEmptyConfig(req ctrl.Request) error {
 	return nil
 }
 
+// filterConfigs filters the given FRRConfiguration such that only the ones matching the given labels are returned.
+// This also validates that the configuration objects have a valid nodeSelector.
+func filterConfigs(cfgs []frrk8sv1beta1.FRRConfiguration, nodeLabels map[string]string) ([]frrk8sv1beta1.FRRConfiguration, error) {
```

Comment:
I'd either rename to `configsForNode` or change the `nodeLabels` parameter 
Also, I am torn on wether this should belong to the translation logic.

Reply:
done